src/SonarScanner.MSBuild.TFS/TrxFileReader.cs

```diff
@@ -82,153 +82,132 @@ public TrxFileReader(ILogger logger)
         /// * resolve the attachment entry to an absolute path</remarks>
-        public string LocateCodeCoverageFile(string buildRootDirectory)
+        public IEnumerable<string> FindCodeCoverageFiles(string buildRootDirectory)
         {
             if (string.IsNullOrWhiteSpace(buildRootDirectory))
             {
                 throw new ArgumentNullException(nameof(buildRootDirectory));
             }
 
-            string coverageFilePath = null;
+            var trxFilePaths = FindTrxFiles(buildRootDirectory);
 
-            var trxFilePath = FindTrxFile(buildRootDirectory);
+            if (!trxFilePaths.Any())
+            {
+                return Enumerable.Empty<string>();
+            }
 
-            if (!string.IsNullOrEmpty(trxFilePath))
+            Debug.Assert(trxFilePaths.All(File.Exists), "Expecting the specified trx files to exist.");
+
+            if (!TryExtractCoverageFilePaths(trxFilePaths, out var attachmentUris))
             {
-                Debug.Assert(File.Exists(trxFilePath), "Expecting the specified trx file to exist: " + trxFilePath);
+                return Enumerable.Empty<string>();
+            }
 
-                coverageFilePath = TryGetCoverageFilePath(trxFilePath);
+            if (!attachmentUris.Any())
+            {
+                this.logger.LogDebug(Resources.TRX_DIAG_NoCodeCoverageInfo);
+                return Enumerable.Empty<string>();
             }
 
-            return coverageFilePath;
+            this.logger.LogDebug(Resources.TRX_DIAG_CodeCoverageAttachmentsFound, string.Join(", ", attachmentUris));
+
+            var potentialCoverageFiles = GetPotentialCoverageFiles(trxFilePaths, attachmentUris)
+                .Distinct()
+                .Where(File.Exists)
+                .ToList();
+
+            if (potentialCoverageFiles.Count == 0)
+            {
+                this.logger.LogWarning(Resources.TRX_WARN_CoverageAttachmentsNotFound);
+            }
+
+            return potentialCoverageFiles;
         }
 
-        public string FindTrxFile(string buildRootDirectory)
+        private IEnumerable<string> GetPotentialCoverageFiles(IEnumerable<string> trxFilePaths, IEnumerable<string> coverageAttachmentPaths)
+        {
+            foreach (var trxPath in trxFilePaths)
+            {
+                var trxDirectoryName = Path.GetDirectoryName(trxPath);
+                var trxFileName = Path.GetFileNameWithoutExtension(trxPath);
+
+                foreach (var coveragePath in coverageAttachmentPaths)
+                {
+                    yield return coveragePath;
+                    yield return Path.Combine(trxDirectoryName, trxFileName, "In", coveragePath);
+                    // https://jira.sonarsource.com/browse/SONARMSBRU-361
+                    // With VSTest task the coverage file name uses underscore instead of spaces.
+                    yield return Path.Combine(trxDirectoryName, trxFileName.Replace(' ', '_'), "In", coveragePath);
```

Comment:
The previous version of the code would emit a warning if the file referred to in the _trx_ couldn't be resolved: the new version will silently ignore the unresolvable file. It would be better to emit a warning in this case.
The approach in the old version in lines 169 would work fine for this.